app/routes/reset.js

```diff
@@ -1,21 +1,15 @@
 import Route from 'ember-route';
-import service from 'ember-service/inject';
-import config from 'wnyc-web-client/config/environment';
+import DeauthenticatedRouteMixin from 'wnyc-web-client/mixins/deauthenticated-route-mixin';
 
-export default Route.extend({
-  config,
-  session: service(),
-  setupController(controller) {
-    controller.set('config', this.get('config'));
-    controller.set('session', this.get('session'));
-    return this._super(...arguments);
-  },
+export default Route.extend(DeauthenticatedRouteMixin, {
   beforeModel(transition) {
     if ( !transition.queryParams.email || !transition.queryParams.confirmation ) {
       // if we got here with missing url parameters,
       // (i.e. typing url, bad copy/paste)
       // send the user to the 'forgot' page so they can request a reset email
       this.transitionTo('forgot');
+    } else {
+      this._super(...arguments);
```

Comment:
I think you do if you want the `beforeModel` method in the mixin to get called. Here's an example I found that uses controllers and `init`, but routes and `beforeModel` should work the same way. https://ember-twiddle.com/b977be3e108dd4a315a84e0dcea8e587?fileTreeShown=false&openFiles=controllers.application.js%2C